geode-core/src/main/java/org/apache/geode/cache/client/internal/QueueManagerImpl.java

```diff
@@ -966,6 +965,17 @@ public void recoverPrimary(Set<ServerLocation> excludedServers) {
     }
   }
 
+  @VisibleForTesting
+  public static boolean isPrimaryRecoveryNeeded(final ConnectionList queueConnectionList) {
```

Comment:
For unit tests in the same package this doesn't need to be public. Then you can also drop the annotation.